### DISCUSSION TITLE
Verel Sanic Python Loop can only be retrieved after the app has started running

### DISCUSSION THREAD

**Discussion by justanotherbyte** [vercel/community#2723] [Help]
### Summary

I'm attempting to run a sanic serverless function, it deploys fine, but when I try to access the function, I get a runtime error like so:

```python
[ERROR] SanicException: Loop can only be retrieved after the app has started running. Not supported with `create_server` function
Traceback (most recent call last):
  File "/var/task/vc__handler__python.py", line 305, in vc_handler
    response = asgi_cycle(__vc_module.app, body)
  File "/var/task/vc__handler__python.py", line 208, in __call__
    loop.run_until_complete(asgi_task)
  File "/var/lang/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/var/task/sanic/app.py", line 1366, in __call__
    self._asgi_app = await ASGIApp.create(self, scope, receive, send)
  File "/var/task/sanic/asgi.py", line 128, in create
    instance.transport.loop = sanic_app.loop
  File "/var/task/sanic/app.py", line 258, in loop
    raise SanicException(
```

Here's my requirements.txt file:
```txt
sanic==23.3.0
asyncpg==0.27.0
```

### Steps to Reproduce

Start a sanic serverless function with the following code:
```python
@app.get("/posts")
async def get_posts(_: Request):
    conn = await connect()
    records = await conn.fetch(
        "SELECT * FROM posts"
    )
    await conn.close()

    records = sorted(records, reverse=True, key=lambda r: r["created_at"])

    posts = []
    for record in records:
        data = serialize_record(record)
        posts.append(data)

    return json(posts, headers=HEADERS)
```

**Comment by amyegan** [comment]
Hi @justanotherbyte. I think this bit about ASGI in our Python Runtime documentation might be related: https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/python#asynchronous-server-gateway-interface

Can you take a look and let me know if that solves the problem for you?

**Comment by Grozoff** [comment]
I faced with the same issue and found that vercel works fine with `sanic 20.12.7` and lesser versions. Unfortunately, I cannot use versions less than `sanic 21.3.0` due to dependencies with other libraries

**Comment by kasperBengtson** [comment]
Thanks for sharing, @Grozoff. This saved me hours. @amyegan maybe you could update the documentation stating what version of Sanic that you support for serverless functions?
